addons/web/static/tests/views/fields/text_field.test.js

```diff
@@ -0,0 +1,346 @@
+    });
+    const textarea = queryOne(".o_field_text textarea");
+    expect(window.getComputedStyle(textarea).resize).toBe("none");
+});
+
+test("should have correct height after onchange", async () => {
+    Product._fields.name = fields.Char({
+        onChange: (record) => {
+            expect.step("onchange");
+            record.description = "Content ".repeat(100); // long text
+        },
+    });
+    Product._records = [{ id: 1, description: "Description as text" }];
+    await mountView({
+        type: "form",
+        resModel: "product",
+        resId: 1,
+        arch: `<form><field name="description"/><field name="name"/></form>`,
+    });
+    const textarea = queryOne(".o_field_text textarea");
+    const initialHeight = textarea.offsetHeight;
+    await fieldInput("name").edit("Let's trigger the onchange");
+    await animationFrame();
+    expect(textarea.offsetHeight).toBeGreaterThan(initialHeight);
+    await fieldTextArea("description").edit("Description as text");
+    expect(textarea.offsetHeight).toBe(initialHeight);
+    expect(["onchange"]).toVerifySteps();
+});
+
+test("it has a correct height in editable list", async () => {
+    Product._records = [{ id: 1, description: "L\no\nn\ng\nD\ne\ns\nc\nr\ni\np\nt\ni\no\nn\n" }];
+    onRpc("/web/dataset/call_kw/res.users/has_group", () => true);
+    await mountView({
+        type: "list",
+        resModel: "product",
+        arch: '<list editable="top"><field name="description"/></list>',
+    });
+    await contains(".o_data_row .o_data_cell").click();
+    const textarea = queryOne(".o_field_text textarea");
+    expect(textarea.clientHeight).toBe(textarea.scrollHeight);
+    await contains("tr:not(.o_data_row)").click();
+    const cell = queryOne(".o_data_row .o_data_cell");
+    expect(cell.clientHeight).toBe(cell.scrollHeight);
+});
+
+test("should resize on reset", async () => {
+    Product._records = [{ id: 1 }];
+    Product._fields.name = fields.Char({
+        onChange: (record) => {
+            expect.step("onchange");
+            record.description = "L\no\nn\ng\nD\ne\ns\nc\nr\ni\np\nt\ni\no\nn\n";
+        },
+    });
+    await mountView({
+        type: "form",
+        resModel: "product",
+        resId: 1,
+        arch: `<form><field name="name"/><field name="description"/></form>`,
+    });
+    await fieldInput("name").edit("Let's trigger the onchange");
+    const textarea = queryOne(".o_field_text textarea");
+    expect(textarea.clientHeight).toBe(textarea.scrollHeight);
+    expect(["onchange"]).toVerifySteps();
+});
+
+test("set row on text fields", async () => {
+    Product._records = [{ id: 1, description: "Description as text" }];
+    await mountView({
+        type: "form",
+        resModel: "product",
+        resId: 1,
+        arch: `<form><field name="description" rows="4"/></form>`,
+    });
+    const textarea = queryOne(".o_field_text textarea");
+    expect(textarea.rows).toBe(4);
```

Comment:
2 fields, same value (one with rows="30"), height of second should be larger that first (+ check rows attr)